### DISCUSSION TITLE
Netbox Custum DropDown menu

### DISCUSSION THREAD

**Discussion by Gebaa** [netbox-community/netbox#16333] [Help Wanted!] [answered]
Hi Folks,

I want to add a new drop-down menu item to the side panel. I am new to the Python Django framework and please guide me to add this customer menu item. 

![image](https://github.com/netbox-community/netbox/assets/95199815/02411d25-c466-47e2-a5d5-729d13b681a9)

Netbox version: 3.7.2

If you need more information, I will provide.

**Comment by markkuleinio** [comment, marked as answer]
You can do that with plugins. Some links:

- https://netboxlabs.com/docs/netbox/en/stable/plugins/development/
- https://github.com/netbox-community/netbox/wiki/Plugins
- https://github.com/markkuleinio/netbox-hello-world-page (my example plugin that has a menu item)

**Comment by Gebaa** [comment]
Hi @markkuleinio ,

Thanks for the solution. I'm new for the Django framework. I followed the steps for installing your pluging (https://github.com/markkuleinio/netbox-hello-world-page) and I can see the expected result of netbox GUI. But I have to do a few customizations on your plugging to apply changes according to my deployment.

I tried to find the installed code of your plugging inside my server netbox location but could not find it.

Can you advise me on which file in which location I need to customize according to my requirements?

**Reply by markkuleinio** [reply]
You can clone the repo somewhere, make modifications, and install the package with "sudo /opt/netbox/venv/bin/pip install -e nameofthedirectory". Or you can just edit the files installed in the venv, they are found somewhere under /opt/netbox/venv/lib/. You need to restart NetBox after any modification.
